### ISSUE TITLE
Issue sorting error bars with confidence intervals

### ISSUE THREAD

**Issue by mc-robinson** [altair-viz/altair#1331]
First, thanks for this library! It has allowed me to do so much in this past week so effortlessly -- the below plot has been the main one giving me tons of trouble. 

In particular, I am trying to create a plot with error bars such as [in the examples](https://altair-viz.github.io/gallery/error_bars_with_ci.html) but with the y-axis sorted by the x-axis. I have produced the following two plots individually:

```
base = alt.Chart(oscar_reviews_df, width=600, height=400)

points = base.mark_point(filled=True).encode(
    x=alt.X(
        'mean(critic_score)',
        scale=alt.Scale(zero=False),
        axis=alt.Axis(title='Mean Rating'),
        
    ),
    y=alt.Y('movie',
            type='nominal',
            sort=alt.EncodingSortField(field='critic_score', op='mean', order='descending')
    ),
    color=alt.value('black')
)

points
```
![image](https://user-images.githubusercontent.com/10458975/52675883-1fc56580-2f20-11e9-9a3e-0eb156863974.png)
and
```
base = alt.Chart(oscar_reviews_df, width=600, height=400)

error_bars = base.mark_rule().encode(
    x=alt.X('ci0(critic_score)',scale=alt.Scale(zero=False)),#,axis=alt.Axis(labels=False, ticks=False)),
    x2=alt.X2('ci1(critic_score)'),
    y=alt.Y('movie',
            type='nominal',
            sort=alt.EncodingSortField(field='critic_score', op='mean', order='descending'),
    ),
)

error_bars
```
![image](https://user-images.githubusercontent.com/10458975/52675849-08867800-2f20-11e9-834e-7e77806fcb90.png)

However, when trying to put the two charts together, the sorting disappears:

```
base = alt.Chart(oscar_reviews_df, width=600, height=400)

points = base.mark_point(filled=True).encode(
    x=alt.X(
        'mean(critic_score)',
        scale=alt.Scale(zero=False),
        axis=alt.Axis(title='Mean Rating'),
        
    ),
    y=alt.Y('movie',
            type='nominal',
            sort=alt.EncodingSortField(field='critic_score', op='mean', order='descending')
    ),
    color=alt.value('black')
)

error_bars = base.mark_rule().encode(
    x=alt.X('ci0(critic_score)',scale=alt.Scale(zero=False)),#,axis=alt.Axis(labels=False, ticks=False)),
    x2=alt.X2('ci1(critic_score)'),
    y=alt.Y('movie',
            type='nominal',
            sort=alt.EncodingSortField(field='critic_score', op='mean', order='descending'),
    ),
)

(points + error_bars) #.resolve_scale(x='independent')
```

![image](https://user-images.githubusercontent.com/10458975/52675943-4d121380-2f20-11e9-82ba-f2cc4338335c.png)

I cannot for the life of me find how to fix this, and would really appreciate some help. And thanks again for all the work you all have done on Altair.

**Comment by jakevdp** [comment]
This is a known issue in Vega-Lite, that will be fixed in the next release (version 3.0). See #820 and https://github.com/vega/vega-lite/issues/2177

**Comment by mc-robinson** [comment]
Great, thanks for the info. I'll close the issue since #820 seems to cover it.
